x-pack/test/api_integration/apis/ml/saved_objects/sync_trained_models.ts

```diff
@@ -0,0 +1,205 @@
+      // check to see if a sync is needed
+      const syncNeeded = await runSyncCheckRequest(USER.ML_POWERUSER_ALL_SPACES, 200);
+      expect(syncNeeded.result).to.eql(false, 'sync should not be needed');
+
+      await ml.api.createDataFrameAnalyticsJob(
+        ml.commonConfig.getDFAIhpOutlierDetectionJobConfig(dfaJobId1),
+        idSpace1
+      );
+      await ml.api.updateJobSpaces(dfaJobId1, 'data-frame-analytics', [idSpace2], [], idSpace1);
+      await ml.api.assertJobSpaces(dfaJobId1, 'data-frame-analytics', [idSpace1, idSpace2]);
+
+      // create model in es, so saved object creation is needed
+      const model1 = getTestModel(modelIdSpace1, 'classification', dfaJobId1);
+      await ml.api.createTrainedModelES(model1.model_id, model1.body);
+
+      // create model via kibana, so saved object creation is no needed
+      const model2 = getTestModel(modelIdSpace2, 'classification', dfaJobId1);
+      await ml.api.createTrainedModel(model2.model_id, model2.body);
+
+      // create model via kibana, but then delete model in es
+      // so saved object needs removal
+      const model3 = getTestModel(modelIdSpace3, 'classification', dfaJobId1);
+      await ml.api.createTrainedModel(model3.model_id, model3.body);
+      await ml.api.deleteTrainedModelES(model3.model_id);
+
+      // check to see if a sync is needed
+      const syncNeeded2 = await runSyncCheckRequest(USER.ML_POWERUSER_ALL_SPACES, 200);
+      expect(syncNeeded2.result).to.eql(true, 'sync should be needed');
+
+      // run the sync request and verify the response
+      const body = await runSyncRequest(USER.ML_POWERUSER_ALL_SPACES, 200);
+
+      expect(body).to.eql({
+        datafeedsAdded: {},
+        datafeedsRemoved: {},
+        savedObjectsCreated: {
+          'trained-model': {
+            [modelIdSpace1]: { success: true },
+          },
+        },
+        savedObjectsDeleted: {
+          'trained-model': { dfa3_classification_model_n_0: { success: true } },
+        },
+      });
+    });
+
+    it('should not sync anything if all objects are already synced', async () => {
+      // check to see if a sync is needed
+      const syncNeeded = await runSyncCheckRequest(USER.ML_POWERUSER_ALL_SPACES, 200);
+      expect(syncNeeded.result).to.eql(false, 'sync should not be needed');
+
+      await ml.api.createDataFrameAnalyticsJob(
+        ml.commonConfig.getDFAIhpOutlierDetectionJobConfig(dfaJobId1),
+        idSpace1
+      );
+      await ml.api.updateJobSpaces(dfaJobId1, 'data-frame-analytics', [idSpace2], [], idSpace1);
+      await ml.api.assertJobSpaces(dfaJobId1, 'data-frame-analytics', [idSpace1, idSpace2]);
+
+      // create model in es, so saved object creation is needed
+      const model1 = getTestModel(modelIdSpace1, 'classification', dfaJobId1);
+      await ml.api.createTrainedModelES(model1.model_id, model1.body);
+
+      // create model via kibana, so saved object creation is no needed
```

Comment:
Nit - should be `not needed`

Reply:
https://github.com/elastic/kibana/pull/130115/commits/a65265454a4be24e54f6d412652fff6638936047